### DISCUSSION TITLE
How to know if an ArcArray is shared

### DISCUSSION THREAD

**Discussion by daniellga** [rust-ndarray/ndarray#1398] [General]
I would like to know if the inner `Arc` of the `ArcArray` is shared. How can I do it?

**Comment by bluss** [comment]
I don't think there is an API for this but it could be added in a PR. Note that any mutable access to the elements of the ArcArray will unshare it (at that point it's known that it is not shared.)

**Reply by daniellga** [reply]
Just did it. Thanks!
